aea/helpers/dependency_tree.py

```diff
@@ -73,14 +73,6 @@ def dump_yaml(
             yaml.dump(data, fp, Dumper=_AEAYamlDumper)
 
 
-def without_hash(public_id: str) -> PublicId:
-    """Convert to public id."""
-    try:
-        return PublicId.from_str(public_id)
-    except ValueError:
-        return PublicId.from_json(PublicId.from_str(public_id).json)
-
-
```

Comment:
First of all, this does not return a public id without hash, it will contain the hash, so it's a complete misnomer. Secondly, this functionality is not used anywhere. And finally, this functionality is covered here:
https://github.com/valory-xyz/open-aea/blob/3146dea7bf2f2dec66ccd3bf2209a7fa0c01df97/aea/configurations/data_types.py#L495-L503

Reply:
Can you check whether it is used in open-autonomy. It's technically a breaking change so we need to be careful about removing it.

Reply:
this is not used in open-autonomy either